src/component-library/NumberInput/NumberInput.style.tsx

```diff
@@ -1,13 +1,7 @@
 import styled from 'styled-components';
 
-import { Input, InputProps } from 'component-library/Input';
+import { Input } from 'component-library/Input';
 
-const BaseNumberInput = styled(Input).attrs<InputProps>({
-  type: 'number',
-  step: 'any',
-  pattern: '[-+]?[0-9]*[.,]?[0-9]+',
```

Comment:
How is this handled using stately?

Reply:
![image](https://user-images.githubusercontent.com/43269067/180193681-9d1ee0a0-4e4c-4c73-87f0-74317f1ef8f4.png)
![image](https://user-images.githubusercontent.com/43269067/180193753-aeb85af1-d520-4b1a-aa70-baec0f872adb.png)

Reply:
`react-stately` - state
`react-aria` - accessbility and semantics

Reply:
That wasn't what I meant :) I meant specifically the line `pattern: '[-+]?[0-9]*[.,]?[0-9]+` - where is the `pattern` attribute being applied?

Reply:
Sorry for the misunderstanding on my end.
For I could gather, they don't apply this input attribute, instead they validate the input programatically.

There are many ways to approach this component. I have searched on many libraries and I have found both approaches, using native attrs and programatically.

This to say that we do not need to use this library in every single component, but I don't dislike at all having it here because they cover a lot of features in this kind of component (which some could be out of our scope)

![image](https://user-images.githubusercontent.com/43269067/180225975-72a826d2-5578-4d48-a60e-c72224a1ec7f.png)

Reply:
Ok, not a big deal - as long as we have a way to enforce the same behaviour, then it's not an issue that it's not done using the pattern attribute :)